force-app/main/default/lwc/deleteSelectedContacts/deleteSelectedContacts.js

```diff
@@ -4,66 +4,117 @@ import start from '@salesforce/apex/ContactController.start';
         if(result.data){
             this.data = result.data;
             this.error = undefined;
+            let preparedContacts = [];
+            this.data.forEach(contact => {
+                let preparedContact = {};
+                preparedContact.Id = contact.Id;
+                preparedContact.nameLink = "/" + contact.Id;
+                preparedContact.AccountId = contact.AccountId;
+                preparedContact.Name = contact.FirstName + " " + contact.LastName;
+                preparedContact.AccountName = contact.Account ? contact.Account.Name : "";               
+                preparedContact.Phone = contact.Phone;
+                preparedContact.Email = contact.Email;
+                preparedContacts.push(preparedContact);
+                let temp = {              
+                    Id : contact.Id,
+                    AccountId: contact.AccountId,
+                    attributes: {
+                        type: "Contact"
+                    }                
+                };
+                this.allContactsMap.set(contact.Id+"", temp);
+            });
+            this.data = preparedContacts;
+            let numberOfPages = Math.ceil(this.data.length / this.perpage);
+            for (let i = 0; i < numberOfPages; i++) {
+                this.pages.push(i);
+                let startIndex = ((i+1)*this.perpage) - this.perpage;
+                let endIndex = ((i+1)*this.perpage);
+                let tmp = this.data.slice(startIndex, endIndex);
+                this.arrayDataPerPage.push(tmp);
+            }
+            this.pageData();
         }else if (result.error){
-            this.error = result.error;
+            this.error = result.error;d
             this.data = undefined;
         }
     }
 
     getSelectedRecords(event) {
-
         const selectedRows = event.detail.selectedRows;
-        this.recordsCount = event.detail.selectedRows.length;
-        let contactIdSet = new Set();
-        for (let i = 0; i < selectedRows.length; i++) {
-            contactIdSet.add(selectedRows[i].Id);            
-        }
-        if(contactIdSet){
-            this.selectedRecords = Array.from(contactIdSet);
+        if(this.selection[this.data] === undefined || selectedRows.length != this.selection[this.data].length){
+            let contactSetId = new Set();
+            for (let i = 0; i < selectedRows.length; i++) {
+            contactSetId.add(selectedRows[i].Id);           
+            }
+        this.selection[this.page] = Array.from(contactSetId);
+        this.recordsCount = this.selection.flat().length;
+        }                     
+    }
+
+    getFinalSelectedContacts(){
+        let contactsId = this.selection.flat();
+        let finalSelectedContacts = [];
+        if(contactsId.length === 0){
+            this.error = new Error("No one contacts are selected!");
+             this.dispatchEvent(
+             new ShowToastEvent({
+                    title: 'Error!!', 
```

Comment:
one exclamation point is more than enough

Reply:
Fixed